bindings/python/src/create_torrent.cpp

```diff
@@ -129,10 +129,24 @@ namespace
         add_files(fs, file, [&](std::string const& i) { return cb(i); }, flags);
     }
 
-    void add_file(file_storage& fs, std::string const& file, std::int64_t size
+    void add_file0(file_storage& fs, string_view const& file, std::int64_t size
+       , file_flags_t const flags, std::time_t md, string_view link)
+    {
+       fs.add_file(std::string(file), size, flags, md, std::string(link));
+    }
+
+    void add_file1(file_storage& fs, bytes const& file, std::int64_t size
        , file_flags_t const flags, std::time_t md, std::string link)
```

Comment:
I notice you left this `link` parameter as `std::string`. is it the case that `std::string` accepts both bytes and unicode strings (converted to utf-8), whereas `string_view` only accepts unicode strings?

Reply:
Yep

Let me know if there's some other way of indicating `str`-only input, other than `string_view`. It's a bit weird to me to use it this way since it's meant as a pointer, unlike `std::string` and `bytes`

Reply:
Right. In particular the builtin `std::string` conversion comes from boost-python, not python itself. It's not clear there's an option to disable it. (IMO it's an impedence mismatch with python 3, and adds to my gripes with boost-python).

I meant I wasn't sure whether the `string_view` converter was *intended* to match `str` only, or just intended for convenience wrapping functions that take `string_view`.

In the latter case, someone might decide to make it match `bytes` one day, to match the non-optional behavior with `std::string`. Then maybe it's better to make a custom C++ `str` class analogous to our custom `bytes` class, specifically intended to only match `str`s from python.